### DISCUSSION TITLE
Unable to Connect to AMF of open5Gs core using srsran

### DISCUSSION THREAD

**Discussion by amitpalkar** [srsran/srsRAN_Project#666] [General Help]
Hi,
  I am running a dockerized version of open5GS using the command ( sudo ./apps/gnb/gnb -c ../configs/gnb_zmq.yml)  When i run the core and gnode i get the below error.  Can someone let me know how to proceed?  Attaching my config files.
[amf.zip](https://github.com/user-attachments/files/15788413/amf.zip)
Would appreciate response.
Regards,
Amit Palkar

==================================================================================

--== srsRAN gNB (commit f3ed07a5a) ==--

Connecting to AMF on 10.53.1.2:38412
Failed to connect SCTP socket to 10.53.1.2:38412. error="Connection timed out" timeout=1387ms
srsGNB ERROR: Failed to create SCTP gateway

==================================================================================

open5gs_5gc  | 06/11 12:37:48.630: [sbi] INFO: NF Service [namf-comm] (../lib/sbi/context.c:1812)
open5gs_5gc  | 06/11 12:37:48.630: [sbi] INFO: nghttp2_server() [http://10.53.1.2]:7777 (../lib/sbi/nghttp2-server.c:414)
open5gs_5gc  | 06/11 12:37:48.630: [amf] INFO: ngap_server() [10.53.1.2]:38412 (../src/amf/ngap-sctp.c:61)
open5gs_5gc  | 06/11 12:37:48.630: [smf] INFO: PFCP associated [127.0.0.7]:8805 (../src/smf/pfcp-sm.c:196)
open5gs_5gc  | 06/11 12:37:48.630: [sctp] INFO: AMF initialize...done (../src/amf/app.c:33)

**Comment by pgawlowicz** [comment]
you changed the AMF port to 7777, so you also need to change it in gnb config: `amf.port`, see https://docs.srsran.com/projects/project/en/latest/user_manuals/source/config_ref.html

**Reply by amitpalkar** [reply]
Hi,
    After adding the value 7777 in the gnb_zmq.yml, amf:
  addr: 10.53.1.2                                                 # The address or hostname of the AMF.
  bind_addr: 10.53.1.1                                            # A local IP that the gNB binds to for>
  **port: 7777                                                         # Port number on AMF.**  

i get the error that AMF port number should be between [20000 - 40000]

My relevant contents of open5gs-5gc.yml (dockerized version of open5gs core) are
amf:
   **sbi:
      server:
         - address: 127.0.0.5
           port: 7777**
     client:
        scp:
         - uri: http://127.0.0.22:7777
   ngap:
     server:
        - address: ${OPEN5GS_IP}

Can you please let me know where i am going wrong.

Regards,
Amit Palkar

**Reply by pgawlowicz** [reply]
Use a port from the range  [20000 - 40000], both in gnb and open5gs configs.

**Reply by amitpalkar** [reply]
Hi @pgawlowicz thanks a lot.  Tried using the port number 38412 instead of 7777 in both the gnb and open5gs configs as given below.  But i am facing the same error.   Below are the configurations for open5gs and gnb yml files.

open5gs config.xml
----------------------
amf:
  sbi:
    server:
      #- address: 127.0.0.5
      - address: ${OPEN5GS_IP}
        #port: 7777
        port: 38412
    client:
      scp:
        - uri: http://127.0.0.22:7777
        
gnb_zmq.yml
---------------
amf:
  addr: 10.53.1.2                                                 # The address or hostname of the AMF.
  bind_addr: 10.53.1.1                                            # A local IP that the gNB binds to for>
  port: 38412                                                      # Port number on AMF.

**Reply by amitpalkar** [reply]
ee210469@pcz-mmalab-2:~/work/opensource_ran/srsran/5G/srsRAN_Project/configs$ route -n
Kernel IP routing table
Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
0.0.0.0         10.1.115.1      0.0.0.0         UG    100    0        0 eno1
10.1.115.0      0.0.0.0         255.255.255.0   U     100    0        0 eno1
10.45.0.0       10.53.1.2       255.255.0.0     UG    0      0        0 br-d32e126b4140
10.53.1.0       0.0.0.0         255.255.255.0   U     0      0        0 br-d32e126b4140
10.53.1.0       0.0.0.0         255.255.255.0   U     0      0        0 br-f7bb8ae0ec84
169.254.0.0     0.0.0.0         255.255.0.0     U     1000   0        0 eno1
172.17.0.0      0.0.0.0         255.255.0.0     U     0      0        0 docker0
172.19.1.0      0.0.0.0         255.255.255.0   U     0      0        0 br-6aac9306bce6
172.19.1.0      0.0.0.0         255.255.255.0   U     0      0        0 br-069322a98f54


Above is output of my route table

**Reply by amitpalkar** [reply]
br-d32e126b4140: flags=4099<UP,BROADCAST,MULTICAST>  mtu 1500
        inet 10.53.1.1  netmask 255.255.255.0  broadcast 10.53.1.255
        inet6 fe80::42:adff:fe59:b9a  prefixlen 64  scopeid 0x20<link>
        ether 02:42:ad:59:0b:9a  txqueuelen 0  (Ethernet)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 101  bytes 19075 (19.0 KB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

br-f7bb8ae0ec84: flags=4099<UP,BROADCAST,MULTICAST>  mtu 1500
        inet 10.53.1.1  netmask 255.255.255.0  broadcast 10.53.1.255
        inet6 fe80::42:86ff:fe94:32ad  prefixlen 64  scopeid 0x20<link>
        ether 02:42:86:94:32:ad  txqueuelen 0  (Ethernet)
        RX packets 7  bytes 532 (532.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 81  bytes 8762 (8.7 KB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

docker0: flags=4099<UP,BROADCAST,MULTICAST>  mtu 1500
        inet 172.17.0.1  netmask 255.255.0.0  broadcast 172.17.255.255
        inet6 fe80::42:a1ff:fe33:9076  prefixlen 64  scopeid 0x20<link>
        ether 02:42:a1:33:90:76  txqueuelen 0  (Ethernet)
        RX packets 387297  bytes 22371689 (22.3 MB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 464458  bytes 2151625801 (2.1 GB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

eno1: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 10.1.115.50  netmask 255.255.255.0  broadcast 10.1.115.255
        inet6 fe80::b782:63f6:e248:a839  prefixlen 64  scopeid 0x20<link>
        ether 6c:4b:90:c6:50:79  txqueuelen 1000  (Ethernet)
        RX packets 8293588  bytes 8944724519 (8.9 GB)
        RX errors 0  dropped 32873  overruns 0  frame 0
        TX packets 6669416  bytes 6372563280 (6.3 GB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
        device interrupt 17  memory 0xb1200000-b1220000

Above is my ifconfig command output

**Reply by pgawlowicz** [reply]
it seems that you have two bridge interfaces using the same IP address `10.53.1.1`, and hence there are two entries for `10.53.1.0` in your routing table. You should remove the interface that is not used.
